src/utils_latency.c

```diff
@@ -176,6 +176,28 @@ void latency_counter_reset (latency_counter_t *lc) /* {{{ */
     return;
 
   cdtime_t bin_width = lc->bin_width;
+  cdtime_t max_bin = (lc->max - 1) / lc->bin_width;
+
+/*
+  If max latency is REDUCE_THRESHOLD times less than histogram's range,
+  then reduce bin width twice. REDUCE_THRESHOLD must be >= 2.
+  Value of 4 is selected to reduce frequent changes of bin width.
+*/
+#define REDUCE_THRESHOLD 4
+  if ((lc->bin_width >= HISTOGRAM_DEFAULT_BIN_WIDTH * 2)
+     && (max_bin < HISTOGRAM_NUM_BINS / REDUCE_THRESHOLD))
+  {
```

Comment:
Ok, when no values was received due to gap then lc->max will be equal 0 and lc->bin_width will reduce down to HISTOGRAM_DEFAULT_BIN_WIDTH, by half of width at each latency_counter_reset() call.

I think that is ok, as we don't know which values will arrive after gap.
If you prefer to save bin width during gap - it will be ok to check lc->num !=0 as proposed by you.

Which the final decision will be about this?